subprojects/plugin-use/src/main/java/org/gradle/plugin/use/internal/PluginDependencyResolutionServices.java

```diff
@@ -98,26 +102,25 @@ public ObjectFactory getObjectFactory() {
     }
 
     public PluginRepositoriesProvider getPluginRepositoriesProvider() {
-        return new PluginRepositoriesProvider() {
-            @Override
-            public List<ArtifactRepository> getPluginRepositories() {
-                RepositoryHandler repositories = getResolveRepositoryHandler();
-                List<ArtifactRepository> list = new ArrayList<ArtifactRepository>(repositories.size());
-                for (ArtifactRepository repository : repositories) {
-                    list.add(new PluginArtifactRepository(repository));
-                }
-                return list;
+        return () -> {
+            RepositoryHandler repositories = getResolveRepositoryHandler();
+            List<ArtifactRepository> list = new ArrayList<>(repositories.size());
+            for (ArtifactRepository repository : repositories) {
+                list.add(new PluginArtifactRepository(repository));
             }
+            return list;
         };
     }
 
-    private static class PluginArtifactRepository implements ArtifactRepositoryInternal, ResolutionAwareRepository {
+    private static class PluginArtifactRepository implements ArtifactRepositoryInternal, ContentFilteringRepository, ResolutionAwareRepository {
         private final ArtifactRepositoryInternal delegate;
         private final ResolutionAwareRepository resolutionAwareDelegate;
+        private final RepositoryContentDescriptorInternal repositoryContentDescriptor;
 
         private PluginArtifactRepository(ArtifactRepository delegate) {
             this.delegate = (ArtifactRepositoryInternal) delegate;
             this.resolutionAwareDelegate = (ResolutionAwareRepository) delegate;
+            repositoryContentDescriptor = this.delegate.createRepositoryDescriptor();
```

Comment:
```suggestion
            this.repositoryContentDescriptor = delegate.createRepositoryDescriptor();
```

Reply:
Added the first `this` but the second one remains, it is needed. Name clash but different types 😉